### PR TITLE
Update telemetry format

### PR DESCRIPTION
**Pull request by lbalmaceda**
### Changes

- Changed telemetry format: use `env` to include the library version as the `react-native-auth0` key.
- Updated dev dependencies for fetch-mock to v7. Fixed snapshots.

### Testing

* [x] This change adds unit test coverage
* [x] This change has been tested on the latest version of the platform/language or why not

### Checklist

* [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [x] All existing and new tests complete without errors
* [x] All active GitHub checks have passed

### CODE REVIEW

File: auth/__tests__/__snapshots__/index.spec.js.snap [hunk 1]
```diff
@@ -12,7 +12,7 @@ exports[`auth change password should return successful response 1`] = `Object {}
 
 exports[`auth change password should send correct payload 1`] = `
 Array [
-  "https://samples.auth0.com/dbconnections/change_password",
+  "https://samples.auth0.com/dbconnections/change_password?auth0Client=eyJuYW1lIjoicmVhY3QtbmF0aXZlLWF1dGgwIiwidmVyc2lvbiI6IjEuMC4wIn0%3D",
```
Comment (joshcanhelp) [inline auth/__tests__/__snapshots__/index.spec.js.snap]: This should be a header, not a URL parameter. URL parameters are for redirects.

File: management/__tests__/__snapshots__/users.spec.js.snap [hunk 2]
```diff
@@ -119,7 +119,7 @@ Object {
 
 exports[`users GET user should send correct payload 1`] = `
 Array [
-  "https://samples.auth0.com/api/v2/users/auth0%7C53b995f8bce68d9fc900099c",
+  "https://samples.auth0.com/api/v2/users/auth0%7C53b995f8bce68d9fc900099c?auth0Client=eyJuYW1lIjoicmVhY3QtbmF0aXZlLWF1dGgwIiwidmVyc2lvbiI6IjEuMC4wIn0%3D",
```
Comment (joshcanhelp) [inline management/__tests__/__snapshots__/users.spec.js.snap]: This should be a header, not a URL parameter. URL parameters are for redirects.

File: networking/__tests__/telemetry.spec.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import defaults from '../telemetry';
 
-describe('telemetry', () => {
+describe('basic telemetry', () => {
   it('should have name', () => {
```
Comment (joshcanhelp) [inline networking/__tests__/telemetry.spec.js]: Shouldn't it check for a version here as well?

Comment (lbalmaceda) [reply networking/__tests__/telemetry.spec.js]: You mean the test case that is below this one? This telemetry file defines name and version for this sdk, and does not include anything else. This is used from the telemetry value generator to obtain these details.

**Pull request merged**